### ISSUE TITLE
Godot 4: Scene assignment to property through editor not saved

### ISSUE THREAD

**Issue by jpew** [godotengine/godot#45697]
**Godot version:**

Godot 4 dev 64bit (built from 1/31)

**OS/device including version:**

Ubuntu 20.04

**Issue description:**

While going through the "Your first game" tutorial I ran into an issue with this step:
https://docs.godotengine.org/en/latest/getting_started/step_by_step/your_first_game.html#main-script
where I changed the following to what I believe is the correct new syntax:

`export(PackedScene) var mob_scene` --> `@export var mob_scene: PackedScene`

I then assigned a scene to the property through the editor. But, when I save the project the assignment is lost. The editor just goes back to displaying [empty] for the property.

I was able to work around the issue by assigning the resource in the script.
`@export var mob_scene: PackedScene = preload("res://Mob.tscn")`

**Steps to reproduce:**
export a packed scene variable and assign a scene through the editor

**Minimal reproduction project:**

[dodge.zip](https://github.com/godotengine/godot/files/5922434/dodge.zip)

**Comment by jsa4000** [comment]
Hi, I face the same problem.
As a workaround use @tool annotation.

**Comment by Calinou** [comment]
Does this occur only in `master` or also in 3.2.3?

**Comment by jpew** [comment]
This works as expected in 3.2.3 with the older export syntax. For the master, is `@export var mob_scene: PackedScene` perhaps incorrect?

**Comment by TheKnittedAcorn** [comment]
I also experienced this issue.

I also built a copy in 3.2.3 and it worked as expected with older export syntax.

Thank you, jpew, for finding a work-around!

This is truly the most irritating of bugs.

**Comment by Blackiris** [comment]
I noticed an issue with the following code in gdscript.cpp _update_exports, the following lines:
```
	if (placeholders.size()) { //hm :(

		// update placeholders if any
		Map<StringName, Variant> values;
		List<PropertyInfo> propnames;
		_update_exports_values(values, propnames);

		for (Set<PlaceHolderScriptInstance *>::Element *E = placeholders.front(); E; E = E->next()) {
			E->get()->update(propnames, values);
		}
	}
```

It seems at line: E->get()->update(propnames, values);
The new value of E->get() is replaced by the default value of the script

To be confirmed though, I will continue looking at it

**Comment by jpew** [comment]
Thanks for taking a look. I guess it may be some time before the devs have a chance to address this relatively minor bug.

**Comment by Blackiris** [comment]
Seems to be same issue as:
#41509
#41690
and maybe #45120

**Comment by Blackiris** [comment]
In `PlaceHolderScriptInstance::update`, by changing this line:
`if (!values.has(n) || values[n].get_type() != E->get().type)`
to:
`if (!values.has(n) && values[n].get_type() != E->get().type)`

The value is changed properly, but something weird happens with the following script:
```
@export var mob_scene: PackedScene # = preload("res://Mob.tscn")
@export var num: int
@export var numDefault: int = 42
@export var pathToNode: NodePath
```
When I let numDefault to 42, after saving, the value is set to 0. But if numDefault value is different than 42, the new value is saved properly...
The 3 others fields seems so work as expected

**Comment by Blackiris** [comment]
The root cause is really because of values[n].get_type() != E->get().type, where E->get().type is Nil as pointed in #45120.

Adding the fix given by https://github.com/godotengine/godot/pull/45120#issuecomment-784715031, changing GDScriptAnalyzer::resolve_class_interface, fixes the issue.

**Comment by Blackiris** [comment]
The issue is fixed in one of the latest Godot 4.0 f7ae5442a607ff1fdd7f63b87ad93d067867aeec
Probably same reason as in issue #41690, and also with the non-exported properties being visible (issue to be create soon).

Therefore, I think this issue can be closed too.

**Comment by akien-mga** [comment]
Thanks for testing!
